api-spec/extensions/transaction/transaction.fragment.yaml

```diff
@@ -1,174 +1,211 @@
-        assets:
-          analytic:
-            title: 1-Band Analytic
-            href: http://cool-sat.com/catalog/collections/cs/items/CS3-20160503_132130_04/analytic-1.tif
-    partialItemProperties:
-      type: object
-      description: allows for partial collections of metadata fields
-      additionalProperties: true
-      properties:
-        datetime:
-          $ref: '#/components/schemas/time'
-tags:
-  - name: Extensions
-    description: >-
-      STAC-specific operations to add, remove, and edit items within WFS3
+openapi: 3.0.1
+info:
+  title: The SpatioTemporal Asset Catalog API + WFS3 + Transaction extension
+  version: 0.7.0
+paths:
+  '/collections/{collectionId}/items':
+    post:
+      summary: add a new feature to a collection
+      description: create a new feature in a specific collection
+      operationId: postFeature
+      tags:
+        - Extensions
+      parameters:
+        - $ref: '#/components/parameters/collectionId'
+      requestBody:
+        content:
+          application/json:
+            schema:
+                oneOf:
+                 - $ref: '#/components/schemas/item'
+                 - $ref: '#/components/schemas/itemCollection'
+      responses:
+        '201':
+          description: Status of the create request.
+          content:
+            application/geo+json:
+              schema:
+                type: string
+            text/html:
+              schema:
+                type: string
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '503':
+          $ref: '#/components/responses/InternalServerError'
+        default:
+          description: An error occurred.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/exception'
+            text/html:
+              schema:
+                type: string
+  '/collections/{collectionId}/items/{featureId}':
+    put:
+      summary: update an existing feature by Id with a complete item definition
+      description: >-
+        Use this method to update an existing feature. Requires the entire
+        GeoJSON  description be submitted.
+      operationId: putFeature
+      tags:
+        - Extensions
+      parameters:
+        - $ref: '#/components/parameters/collectionId'
+        - $ref: '#/components/parameters/featureId'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/item'
+      responses:
+        '200':
+          description: Status of the update request.
+          content:
+            text/html:
+              schema:
+                type: string
+            application/json:
+              schema:
+                type: string
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '503':
+          $ref: '#/components/responses/InternalServerError'
```

Comment:
503 is Service Unavailable. InternalServerError is 500. Maybe just go with 5XX (yes, that's actually allowed) to allow all 500-599 errors. Many can occur on servers and should send a harmonized error response.

Reply:
ha, my typo.  5XX sounds great -- I mostly wanted to clearly specify what what the "you submitted an invalid request" and "hit a bug/data store down/etc" cases.  Updated.